easybuild/tools/toolchain/toolchain.py

```diff
@@ -108,6 +108,7 @@ def __init__(self, name=None, version=None, mns=None, class_constants=None, tcde
 
         # toolchain instances are created before initiating build options sometimes, e.g. for --list-toolchains
         self.dry_run = build_option('extended_dry_run', default=False)
+        self.hide_deps = build_option('hide_deps', default=None)
```

Comment:
I would use an empty list as a fallback here (not as default value for the build option, that's something else)

so:

```python
self.hide_deps = build_option('hide_deps', default=None) or []
```